code/controllers/Processes/lighting.dm

```diff
@@ -1,18 +1,86 @@
-var/global/datum/controller/process/lighting/lighting_controller
+// Solves problems with lighting updates lagging shit
+// Max constraints on number of updates per doWork():
+#define MAX_LIGHT_UPDATES_PER_WORK   100
+#define MAX_CORNER_UPDATES_PER_WORK  1000
+#define MAX_OVERLAY_UPDATES_PER_WORK 2000
 
+/var/list/lighting_update_lights    = list()    // List of lighting sources  queued for update.
+/var/list/lighting_update_corners   = list()    // List of lighting corners  queued for update.
+/var/list/lighting_update_overlays  = list()    // List of lighting overlays queued for update.
+
+/var/list/lighting_update_lights_old    = list()    // List of lighting sources  currently being updated.
+/var/list/lighting_update_corners_old   = list()    // List of lighting corners  currently being updated.
+/var/list/lighting_update_overlays_old  = list()    // List of lighting overlays currently being updated.
+
+
+/datum/controller/process/lighting
 /datum/controller/process/lighting/setup()
 	name = "lighting"
-	schedule_interval = LIGHTING_INTERVAL
-	start_delay = 1
-	lighting_controller = src
 
-	create_lighting_overlays()
+	schedule_interval = world.tick_lag // run as fast as you possibly can
+	create_all_lighting_overlays()
```

Comment:
Set a global variable like `lighting_overlays_initialized` to true, see below why.